File: bm-gateway/src/main/kotlin/ru/blimfy/gateway/api/user/dto/UsernameDto.kt
package ru.blimfy.gateway.api.user.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

/**
 * DTO с информацией для обновления имени пользователя.
 *
 * @property username новое имя пользователя.
 * @property password пароль пользователя.
 * @author Владислав Кузнецов.
 * @since 0.0.1.
 */
data class UsernameDto(
    @Size(min = 2, max = 32, message = "Usernames must be between 2 and 32 characters long")
    val username: String,

    @NotBlank(message = "Password is mandatory")
    val password: String,
)

File: ui/src/api/dto/request/auth/UsernameDto.ts
/**
 * DTO с информацией для авторизации существующего пользователя.
 *
 * @author Владислав Кузнецов.
 * @since 0.0.1.
 */
export class UsernameDto {
  /**
   * Имя пользователя.
   */
  readonly username: string = ""

  constructor(username: string) {
    this.username = username;
  }
}
